### ISSUE TITLE
onnxmltools==1.10.0 installation error in python 3.10.0

### ISSUE THREAD

**Issue by hanzigs** [onnx/onnxmltools#537]
Hi,
I am upgrading my python to 3.10.0, currently using 3.9.5 where onnxmltools==1.7.0 working good.
I installed 1.10.0 in py3.10.0, I am getting following error while installing
May I have some help please
```
Collecting onnxmltools==1.10.0
  Using cached onnxmltools-1.10.0-py2.py3-none-any.whl (300 kB)
Collecting skl2onnx
  Using cached skl2onnx-1.11-py2.py3-none-any.whl (274 kB)
Requirement already satisfied: numpy in c:\users\pg\anaconda3\envs\py3100t2\lib\site-packages (from onnxmltools==1.10.0) (1.21.5)
Collecting onnx
  Using cached onnx-1.11.0.tar.gz (9.9 MB)
Requirement already satisfied: protobuf>=3.12.2 in c:\users\pg\anaconda3\envs\py3100t2\lib\site-packages (from onnx->onnxmltools==1.10.0) (3.19.4)
Requirement already satisfied: typing-extensions>=3.6.2.1 in c:\users\pg\anaconda3\envs\py3100t2\lib\site-packages (from onnx->onnxmltools==1.10.0) (4.1.1)
Requirement already satisfied: scipy>=1.0 in c:\users\pg\anaconda3\envs\py3100t2\lib\site-packages (from skl2onnx->onnxmltools==1.10.0) (1.8.0)
Requirement already satisfied: scikit-learn>=0.19 in c:\users\pg\anaconda3\envs\py3100t2\lib\site-packages (from skl2onnx->onnxmltools==1.10.0) (1.0.2)
Collecting onnxconverter-common>=1.7.0
  Using cached onnxconverter_common-1.9.0-py2.py3-none-any.whl (78 kB)
Requirement already satisfied: threadpoolctl>=2.0.0 in c:\users\pg\anaconda3\envs\py3100t2\lib\site-packages (from scikit-learn>=0.19->skl2onnx->onnxmltools==1.10.0) (3.1.0)
Requirement already satisfied: joblib>=0.11 in c:\users\pg\anaconda3\envs\py3100t2\lib\site-packages (from scikit-learn>=0.19->skl2onnx->onnxmltools==1.10.0) (1.1.0)
Building wheels for collected packages: onnx
  Building wheel for onnx (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10\\setup.py'"'"'; __file__='"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\pg\AppData\Local\Temp\pip-wheel-8ukw42_4'
       cwd: C:\Users\pg\AppData\Local\Temp\pip-install-wy5rosfe\onnx_cd00ece8f9974594812aa94a6c991c10\
  Complete output (59 lines):
  fatal: not a git repository (or any of the parent directories): .git
  running bdist_wheel
  running build
  running build_py
  running create_version
  running cmake_build
  Using cmake args: ['C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\Scripts\\cmake.exe', '-DPYTHON_INCLUDE_DIR=C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\include', '-DPYTHON_EXECUTABLE=C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\python.exe', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cp310-win_amd64.pyd', '-DCMAKE_BUILD_TYPE=Release', '-DPY_VERSION=3.10', '-A', 'x64', '-T', 'host=x64', '-DONNX_ML=1', 'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10']
  -- Building for: NMake Makefiles
  CMake Error at CMakeLists.txt:17 (project):
    Generator

      NMake Makefiles

    does not support platform specification, but platform

      x64

    was specified.


  CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
  CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
  -- Configuring incomplete, errors occurred!
  See also "C:/Users/pg/AppData/Local/Temp/pip-install-wy5rosfe/onnx_cd00ece8f9974594812aa94a6c991c10/.setuptools-cmake-build/CMakeFiles/CMakeOutput.log".
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\pg\AppData\Local\Temp\pip-install-wy5rosfe\onnx_cd00ece8f9974594812aa94a6c991c10\setup.py", line 336, in <module>
      setuptools.setup(
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\setuptools\__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\core.py", line 148, in setup
      dist.run_commands()
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\wheel\bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\command\build.py", line 135, in run
      self.run_command(cmd_name)
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\pg\AppData\Local\Temp\pip-install-wy5rosfe\onnx_cd00ece8f9974594812aa94a6c991c10\setup.py", line 232, in run
      self.run_command('cmake_build')
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\pg\AppData\Local\Temp\pip-install-wy5rosfe\onnx_cd00ece8f9974594812aa94a6c991c10\setup.py", line 218, in run
      subprocess.check_call(cmake_args)
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\subprocess.py", line 369, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\Scripts\\cmake.exe', '-DPYTHON_INCLUDE_DIR=C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\include', '-DPYTHON_EXECUTABLE=C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\python.exe', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cp310-win_amd64.pyd', '-DCMAKE_BUILD_TYPE=Release', '-DPY_VERSION=3.10', '-A', 'x64', '-T', 'host=x64', '-DONNX_ML=1', 'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10']' returned non-zero exit status 1.
  ----------------------------------------
  ERROR: Failed building wheel for onnx
  Running setup.py clean for onnx
Failed to build onnx
Installing collected packages: onnx, onnxconverter-common, skl2onnx, onnxmltools
    Running setup.py install for onnx ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10\\setup.py'"'"'; __file__='"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\pg\AppData\Local\Temp\pip-record-0s12drig\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\pg\anaconda3\envs\py3100t2\Include\onnx'
         cwd: C:\Users\pg\AppData\Local\Temp\pip-install-wy5rosfe\onnx_cd00ece8f9974594812aa94a6c991c10\
    Complete output (60 lines):
    fatal: not a git repository (or any of the parent directories): .git
    running install
    running build
    running build_py
    running create_version
    running cmake_build
    Using cmake args: ['C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\Scripts\\cmake.exe', '-DPYTHON_INCLUDE_DIR=C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\include', '-DPYTHON_EXECUTABLE=C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\python.exe', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cp310-win_amd64.pyd', '-DCMAKE_BUILD_TYPE=Release', '-DPY_VERSION=3.10', '-A', 'x64', '-T', 'host=x64', '-DONNX_ML=1', 'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10']
    CMake Error at CMakeLists.txt:17 (project):
      Generator

        NMake Makefiles

      does not support platform specification, but platform

        x64

      was specified.


    CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
    CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
    -- Configuring incomplete, errors occurred!
    See also "C:/Users/pg/AppData/Local/Temp/pip-install-wy5rosfe/onnx_cd00ece8f9974594812aa94a6c991c10/.setuptools-cmake-build/CMakeFiles/CMakeOutput.log".
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\pg\AppData\Local\Temp\pip-install-wy5rosfe\onnx_cd00ece8f9974594812aa94a6c991c10\setup.py", line 336, in <module>
        setuptools.setup(
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\setuptools\__init__.py", line 153, in setup
        return distutils.core.setup(**attrs)
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\core.py", line 148, in setup
        dist.run_commands()
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\setuptools\command\install.py", line 61, in run
        return orig.install.run(self)
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\command\install.py", line 568, in run
        self.run_command('build')
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\command\build.py", line 135, in run
        self.run_command(cmd_name)
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Users\pg\AppData\Local\Temp\pip-install-wy5rosfe\onnx_cd00ece8f9974594812aa94a6c991c10\setup.py", line 232, in run
        self.run_command('cmake_build')
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Users\pg\AppData\Local\Temp\pip-install-wy5rosfe\onnx_cd00ece8f9974594812aa94a6c991c10\setup.py", line 218, in run
        subprocess.check_call(cmake_args)
      File "C:\Users\pg\anaconda3\envs\py3100t2\lib\subprocess.py", line 369, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\Scripts\\cmake.exe', '-DPYTHON_INCLUDE_DIR=C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\include', '-DPYTHON_EXECUTABLE=C:\\Users\\pg\\anaconda3\\envs\\py3100t2\\python.exe', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cp310-win_amd64.pyd', '-DCMAKE_BUILD_TYPE=Release', '-DPY_VERSION=3.10', '-A', 'x64', '-T', 'host=x64', '-DONNX_ML=1', 'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10']' returned non-zero exit status 1.
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\pg\anaconda3\envs\autoMLpy3100t2\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10\\setup.py'"'"'; __file__='"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-wy5rosfe\\onnx_cd00ece8f9974594812aa94a6c991c10\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\pg\AppData\Local\Temp\pip-record-0s12drig\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\pg\anaconda3\envs\py3100t2\Include\onnx' Check the logs for full command output.
```

**Comment by xadupre** [comment]
onnx, onnxruntime are now available on python 3.10.

**Comment by xadupre** [comment]
Closing the issue, feel free to reopen it.
